lib/rendition.js

```diff
@@ -40,33 +40,33 @@ const Metadata = {
     TIFF_IMAGEHEIGHT: 'tiff:imageHeight'
 };
 
-// function fileHash(filename, algorithm) {
-//     return new Promise((resolve, reject) => {
-//         try {
-//             const shasum = crypto.createHash(algorithm);
-
-//             const stream = fs.createReadStream(filename);
-//             stream.on('data', data => shasum.update(data));
-//             stream.on('end', () => resolve(shasum.digest('hex')));
-//             stream.on('error', err => {
-//                 reject(`creating ${algorithm} hash failed: ${err.message || err}`);
-//             });
-//         } catch (error) {
-//             return reject(`creating ${algorithm} hash failed: ${error.message || error}`);
-//         }
-//     });
-// }
+function fileHash(filename, algorithm) {
+    return new Promise((resolve, reject) => {
+        try {
+            const shasum = crypto.createHash(algorithm);
+
+            const stream = fs.createReadStream(filename);
+            stream.on('data', data => shasum.update(data));
+            stream.on('end', () => resolve(shasum.digest('hex')));
+            stream.on('error', err => {
+                reject(`creating ${algorithm} hash failed: ${err.message || err}`);
+            });
+        } catch (error) {
+            return reject(`creating ${algorithm} hash failed: ${error.message || error}`);
+        }
+    });
+}
 
 /**
  * Rendition abstraction
  */
 class Rendition {
-    constructor(instructions, directory, index = 0) {
+    constructor(instructions, directory, index = 0, isPipelineRendition = true) {
         this.instructions = instructions; // e.g. quality, dpi, format etc
         this.directory = directory;
         this.index = index;
         this.target = instructions.target;
```

Comment:
I added this in transformer-image-magick to make it backwards compatible since we don't require `fmt` defined in the pipeline: https://git.corp.adobe.com/nui/transformer-image-magick/blob/master/lib/sdk/rendition.js#L68-L73
could you add this?